backend/helper_zero/views/search.py

```diff
@@ -19,59 +19,71 @@ def list(self, request):
-            org_id__in=org_id_list,
-            item_type=item_type,
+        org_search_results = _get_org_search_results(
+            zipcode, lat, lon, radius, delivery_values, offset
         )
-        donation_requests_ordered_list = DonationRequestSerializer(donation_requests, many=True).data
-
-        response = _build_search_response(org_ordered_list, donation_requests_ordered_list)
+        response = _build_search_response(org_search_results, item_type)
         return Response(response)
 
 
-def _build_search_response(org_ordered_list, donation_requests_ordered_list):
-    org_id_to_donation_requests = defaultdict(list)
-    for donation_request in donation_requests_ordered_list:
-        org_id = donation_request['org_id']
-        org_id_to_donation_requests[org_id] += donation_request
-
-    response = {}
-    for org in org_ordered_list:
-        org_id = org['id']
-        response_info = dict()
-        response_info['org'] = org
-        response_info['donation_requests'] = org_id_to_donation_requests[org_id]
-        response[org_id] = response_info
+def _build_search_response(org_list_results, item_type):
+    response = []
+    for org in org_list_results:
+        if item_type:
+            org['donation_requests'] = list(filter(lambda x: (x['item_type'] == item_type), org['donation_requests']))
+        response += [org]
     return response
 
 
-def _get_organizations(zipcode, lat, lon, radius, delivery_values):
+def _get_org_search_results(zipcode, lat, lon, radius, delivery_values, offset):
     if zipcode:
-        organizations = Organization.objects.filter(
+        org_query_set = Organization.objects.filter(
             zipcode=zipcode,
             is_dropoff_only__in=delivery_values,
         ).order_by('name')
-    else:
-        # Otherwise, use lat and lon
+    elif lat and lon:
         tl_lon, tl_lat, br_lon, br_lat = get_search_bounding_box(lat, lon, radius)
-        organizations = Organization.objects.filter(
+        org_query_set = Organization.objects.filter(
             lat__lte=tl_lat,
             lat__gte=br_lat,
             lon__lte=tl_lon,
             lon__gte=br_lon,
             is_dropoff_only__in=delivery_values,
         ).order_by('name')
+    else:
+        # If no location data, default to showing any orgs that have open requests, capping at 20
+        org_query_set = Organization.objects.filter(
+            donation_requests__isnull=True
+        )[:20]
+
+    org_search_results = OrganizationSerializer(org_query_set, many=True).data
+    if lat and lon:
+        # Order by closest to search location
+        org_search_results.sort(key=lambda x: _get_distance(float(x['lat']), lat, float(x['lon']), lon))
+
+    org_search_results = _paginate_results(org_search_results, offset)
+    return org_search_results
+
+
+def _paginate_results(org_search_results, offset):
+    paginator = Paginator(org_search_results, 20)  # Show 20 orgs per page
```

Comment:
what do you think of allowing the frontend to define how many results they want to show on a page?

Reply:
yeah i was considering that, but couldn't think of when/why we would need to do that so just put a default in for now; we can fix later or lmk if you feel super strongly

Reply:
👍 thinking it'd be useful if we wanted it to be mobile-accessible but not something to worry about for now